Refactor run/primitives page (#293)
Qiskit/documentation: docs/run/primitives.mdx

# Introduction to primitives

> Computing systems are built upon multiples layers of **abstraction**. Abstractions generalize certain

Reviewer: ```suggestion
Computing systems are built upon multiple layers of **abstraction**. Abstractions generalize certain
```

details of *how* the technology works, to allow us to focus our attention on the details that are relevant for a
specific task. The level of abstraction we choose to work with will depend on how we want to interact with the device.
The closer we get to the hardware, the lower-level we need the abstractions to be (for example, we could

> want to manipulate electrical signals), and vice-versa, the more complex the task we want to perform,

Author: ```suggestion
want to manipulate electrical signals), and vice versa, the more complex the task we want to perform,
```

access advanced hardware capabilities of a given provider. For example, with Qiskit Runtime primitives,
you can leverage the latest advancements in error mitigation and suppression by toggling options such as
`optimization_level` and `resilience_level`, rather than building your own implementation of these techniques.

For hardware providers, implementing primitives natively means you can provide your users with a more “out-of-the-box”
way to access your hardware features. It is therefore easier for your users to benefit from your hardware's
best capabilities.

## Estimator

The Estimator primitive computes expectation values of observables with respect to states prepared by quantum circuits.
The Estimator receives circuit-observable pairs (with the observable expressed as a
weighted sum of Pauli operators) as inputs, and returns the computed expectation values per pair, as well as their
variances. Different Estimator implementations support various configuration options. The circuits
can be parametrized, as long as the parameter values are also provided as input to the primitive.

## Sampler

> The Sampler primitive samples results from a series of measurements performed on states prepared by quantum circuits.

Reviewer 2: The "series of measurements" is unclear. I suggest instead:
```suggestion
The Sampler primitive samples results from the classical output registers resulting from execution of quantum circuits.
```

2. `BaseSampler` and `BaseEstimator`

These are abstract base classes that define a common interface for implementing primitives. All other classes in the `qiskit.primitives` module inherit from these base classes, and developers should use these if they are interested in developing their own primitives-based execution model for a specific backend provider. These classes may also be useful for those who want to do highly customized processing and find the existing primitives implementations too simple for their needs.

3. `BackendSampler` and `BackendEstimator`

If a provider does not support primitives natively, you can use these classes to “wrap” any backend into a primitive. Users can write primitive-style code for providers that don’t yet have a primitives-based interface. These classes can be used just like the regular Sampler and Estimator, except they should be initialized with an additional backend argument for selecting which backend to run on.

The Qiskit Runtime primitives provide a more sophisticated implementation (such as with error mitigation) as a cloud-based service.

## Next steps

> <Admonition type="tip" title="Recommendations">

Reviewer 3: Can you add a link to the "get started with primitives" page in the next steps section please?

Author: I set up a suggested addition for that link here https://github.com/Qiskit/documentation/pull/293/files#r1391373158

>     - Review detailed [primitives examples.](primitives-examples)

Author: ```suggestion
    - Read [Get started with primitives](primitives-get-started) to implement primitives in your work.
    - Review detailed [primitives examples.](primitives-examples)
```
